src/main/java/com/shapesecurity/functional/Pair.java

```diff
@@ -70,6 +70,16 @@ public B right() {
         return new Pair<>(this.right, this.left);
     }
 
+    @Nonnull
+    public <A1, B1> Pair<A1, B1> map(@Nonnull F<A, A1> fA, @Nonnull F<B, B1> fB) {
+        return new Pair<>(fA.apply(this.left), fB.apply(this.right));
+    }
+
+    @Nonnull
+    public <T> T map(@Nonnull F2<A, B, T> f) {
```

Comment:
This is just function application. It doesn't need a method. Just do `f(pair.left, pair.right)`.